publish_courses/tests/jasmine/services/PublishCoursesAPIServiceSpec.js

```diff
@@ -1,43 +1,80 @@
+  var pcapi, $httpBackend, $http, $log, $q;
+  var djangoUrl;  // required by pcapi; mock these
+
+  beforeEach(module(function mockDjangoUrl($provide) {
+    djangoUrl = {
+      reverse: function (viewName, args) { return 'fake-url' }
+    };
+    $provide.value('djangoUrl', djangoUrl)
+  }));
+
+  beforeEach(function setupTestEnvironment() {
+    module('PublishCoursesAPIModule');
+    inject(function (_$httpBackend_, _pcapi_, _$http_, _$log_, _$q_) {
+      pcapi = _pcapi_;
+      $httpBackend = _$httpBackend_;
+      $http = _$http_;
+      $log = _$log_;
+      $q = _$q_;
     });
+  });
+
+  beforeEach(function() {
+    // save references to injected services in `this` (jasmine user context)
+    // so they can be accessed by spec helpers
+    this.$httpBackend = $httpBackend;
+    this.getParamsAndRespondWith = spec.getParamsAndRespondWith;
+  });
 
-    describe('Sanity check on dependency injection', function() {
-        it('injects the providers we requested', function () {
-            [$httpBackend, $http, $q, $log, pcapi].forEach(function (thing) {
-                expect(thing).not.toBeUndefined();
-                expect(thing).not.toBeNull();
-            });
-        });
+  afterEach(spec.verifyNoOutstanding);
+
+  describe('Sanity check on dependency injection', function () {
+    it('injects the providers we requested', function () {
+      spec.diSanityCheck([$httpBackend, $http, $q, $log, pcapi]);
     });
+  });
+
+  describe('getCourseSummary', function () {
+    it('GETs course summary with expected params', function () {
+      var accountId = 1;
+      var termId = 2;
+      var actualData = null;
+      var apiResponse = {test: 'ok'};
 
-    describe('getCourseSummary', function() {
-        it('GETs course summary with expected params');
-        it('cancels pending request');
-        it('returns data on success');
-        it('resets pending request tracker on success');
-        it('stops error propagation for canceled pending requests');
-        it('reports status text on non-canceled error');
+      pcapi.CourseSummary.get(accountId, termId).then(function (data) {
+        actualData = data;
+      });
+
+      var options = {method: 'GET', response: apiResponse};
+
+      var actualParams = this.getParamsAndRespondWith(options);
+
+      expect(actualData.test).toBe('ok');
+      // $httpBackend decodes URI to get values, all have been cast to String
+      expect(actualParams).toEqual(
+        {account_id: String(accountId), term_id: String(termId)});
     });
-    describe('createJob', function() {
-        it('POSTs job with expected params');
-        it('returns async promise that resolves on success with data');
-        it('returns status text on error');
+    it('cancels pending request');
+    it('returns data on success');
+    it('resets pending request tracker on success');
+    it('stops error propagation for canceled pending requests');
+    it('reports status text on non-canceled error');
```

Comment:
It amounts to the same thing -- they show up as skipped tests. I prefer this style because it shows they are unimplemented, versus explicitly skipped. I'll leave it like this for now, we can talk about style and consistency offline if we want to change this convention.